arithmetization/src/main/java/net/consensys/linea/zktracer/module/limits/precompiles/RipeMd160Blocks.java

```diff
@@ -80,39 +87,43 @@ public static long gasCost(final Hub hub) {
   public void tracePreOpcode(MessageFrame frame) {
     final OpCode opCode = hub.opCode();
 
-    switch (opCode) {
-      case CALL, STATICCALL, DELEGATECALL, CALLCODE -> {
-        final Address target = Words.toAddress(frame.getStackItem(1));
-        if (target.equals(Address.RIPEMD160)) {
-          final long dataByteLength = hub.transients().op().callDataSegment().length();
-          if (dataByteLength == 0) {
-            return;
-          } // skip trivial hash TODO: check the prover does skip it
-          final int blockCount =
-              (int)
-                      (dataByteLength * 8
-                          + RIPEMD160_ND_PADDED_ONE
-                          + RIPEMD160_LENGTH_APPEND
-                          + (RIPEMD160_BLOCKSIZE - 1))
-                  / RIPEMD160_BLOCKSIZE;
-
-          final long wordCount = (dataByteLength + 31) / 32;
-          final long gasNeeded = PRECOMPILE_BASE_GAS_FEE + PRECOMPILE_GAS_FEE_PER_EWORD * wordCount;
-
-          if (hub.transients().op().gasAllowanceForCall() >= gasNeeded) {
-            this.counts.push(this.counts.pop() + blockCount);
-          }
+    if (opCode.isCall()) {
+      final Address target = Words.toAddress(frame.getStackItem(1));
+      if (target.equals(Address.RIPEMD160)) {
+        final long dataByteLength = hub.transients().op().callDataSegment().length();
+
+        if (dataByteLength == 0) {
+          return;
+        } // skip trivial hash TODO: check the prover does skip it
```

Comment:
did we check it ?

Reply:
The prover should only prove what we ask it to prove. Do we need to check anything ?